src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/builder/BigQuerySQLBuilder.java

```diff
@@ -24,33 +24,90 @@
 import io.cdap.cdap.etl.api.join.JoinKey;
 import io.cdap.cdap.etl.api.join.JoinStage;
 
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
 /**
- * Helper class used to generate BigQuery SQL Statements
+ * Helper class used to generate BigQuery SQL Statements for Joins.
+ * <p>
+ * Many methods in this class have proteced visibility for the purposes of testing.
  */
 public class BigQuerySQLBuilder {
+
+  /**
+   * Allows us to find a table name in the user supplied advanced SQL Join expression.
+   * Explanation of groups:
+   * 1. Matches on 3 possibilities:
+   *   a. The beggining of the string,
+   *   b. A non-word character (like a parenthesys or symbol)
+   *   c. Ensures that the preceding character is not a dot with one or several spaces following it, as this would
+   *      mean this is a column name and not a table name
+   * 2. Optionally, match a backtick (used for quoted identifiers in Spark)
+   * 3. Match the table name or alias in the statement.
+   * 4. Optionally, match a backtick (used for quoted identifiers in Spark)
+   * 5. Match 0 or more whitespace characters
+   * 6. Match a dot character, as this means a column name follows.
+   * <p>
+   * This can match expressions such as:
+   * table.column
+   * `table`.column
+   * table .column
+   * `table` .column
+   * table    .  column
+   * `table`    .  column
+   */
+  private static final String TABLE_NAME_PATTERN = "(^|\\W|^\\.\\s*)(`?)(%s)(`?)(\\s*)(\\.)";
```

Comment:
`^\\.\\s*` does not mean "not a dot". It means start of line followed by dot. "^" would mean not only inside square brackets.
Please add separate unit tests to ensure the pattern does what's expected

Reply:
Oof, good catch.

Reply:
Will add further regex tests

Reply:
I have removed the regex based on Albert's comments.